modules/azure/frontdoor.go

```diff
@@ -0,0 +1,127 @@
+// This function would fail the test if there is an error.
+func FrontDoorInstanceExists(t testing.TestingT, instanceName string, resourceGroupName string, subscriptionID string) bool {
+	exists, err := FrontDoorInstanceExistsE(instanceName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// GetFrontDoorInstance gets a Front Door instance by name if it exists for the subscription.
+// This function would fail the test if there is an error.
+func GetFrontDoorInstance(t testing.TestingT, instanceName string, resourceGroupName string, subscriptionID string) *frontdoor.FrontDoor {
+	fd, err := GetFrontDoorInstanceE(instanceName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return fd
+}
+
+// FrontendEndpointExistsForFrontDoorInstance indicates whether the frontend endpoint instance exists for the Front Door instance.
+// This function would fail the test if there is an error.
+func FrontendEndpointExistsForFrontDoorInstance(t testing.TestingT, endpointName string, instanceName string, resourceGroupName string, subscriptionID string) bool {
+	exists, err := FrontendEndpointExistsE(endpointName, instanceName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// GetFrontendEndpointsForFrontDoorInstance gets a frontend endpoint by name for a Front Door instance if it exists for the subscription.
+// This function would fail the test if there is an error.
+func GetFrontendEndpointForFrontDoorInstance(t testing.TestingT, endpointName string, instanceName string, resourceGroupName string, subscriptionID string) *frontdoor.FrontendEndpoint {
+	ep, err := GetFrontendEndpointForFrontDoorInstanceE(endpointName, instanceName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return ep
+}
+
+// FrontDoorInstanceExistsE indicates whether the Front Door instance exists and may return an error.
+func FrontDoorInstanceExistsE(instanceName string, resourceGroupName string, subscriptionID string) (bool, error) {
+	_, err := GetFrontDoorInstanceE(instanceName, resourceGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// FrontendEndpointExistsE indicates whether the endpoint exists for the Front Door instance and may return an error.
+func FrontendEndpointExistsE(endpointName string, instanceName string, resourceGroupName string, subscriptionID string) (bool, error) {
+	_, err := GetFrontendEndpointForFrontDoorInstanceE(endpointName, instanceName, resourceGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// GetFrontDoorInstanceE gets the Front Door instance if it exists and may return an error.
+func GetFrontDoorInstanceE(instanceName, resoureGroupName, subscriptionID string) (*frontdoor.FrontDoor, error) {
```

Comment:
rename parameter  "istancename" to frontDoorName to match the naming style in the rest on the azure module helpers. this applies to all functions in file as well  https://github.com/gruntwork-io/terratest/blob/e19192d2a19b27f52bb368bce54ccefa8fd8cd27/modules/azure/loganalytics.go#L14
```suggestion
func GetFrontDoorE(frontDoorName, resoureGroupName, subscriptionID string) (*frontdoor.FrontDoor, error)
```